draft-schlesinger-cfrg-act.md

```diff
@@ -391,44 +398,60 @@ IssueResponse(sk, request, c):
     - InvalidIssuanceRequestProof, raised when the client proof verification fails
 
   Steps:
-    1. Parse request as (K, gamma, k_bar, r_bar)
-    2. // Verify proof of knowledge
-    3. K1 = H2 * k_bar + H3 * r_bar - K * gamma
-    4. transcript = CreateTranscript("request")
-    5. AddToTranscript(transcript, K)
-    6. AddToTranscript(transcript, K1)
-    7. if GetChallenge(transcript) != gamma:
-    8.     raise InvalidIssuanceRequestProof
-    9. // Create BBS signature on (c, k, r)
-    10. e <- Zq
-    11. A = (G + H1 * c + K) * (1/(e + sk))  // K = H2 * k + H3 * r
-    12. // Generate proof of correct computation
-    13. alpha <- Zq
-    14. Y_A = A * alpha
-    15. Y_G = G * alpha
-    16. X_A = G + H1 * c + K
-    17. X_G = G * e + pk
-    18. transcript_resp = CreateTranscript("respond")
-    19. AddToTranscript(transcript_resp, c)
-    20. AddToTranscript(transcript_resp, e)
-    21. AddToTranscript(transcript_resp, A)
-    22. AddToTranscript(transcript_resp, X_A)
-    23. AddToTranscript(transcript_resp, X_G)
-    24. AddToTranscript(transcript_resp, Y_A)
-    25. AddToTranscript(transcript_resp, Y_G)
-    26. gamma_resp = GetChallenge(transcript_resp)
-    27. z = gamma_resp * (sk + e) + alpha
-    28. response = (A, e, gamma_resp, z, c)
-    29. return response
+    // Verify proof of knowledge
+    1. Parse request as (K, pok)
+    2. relation = build_relation_issue_request(H2, H3, K)
+    3. nizk = NISigmaProtocol("request", relation)
+    4. if nizk.verify(pok) == false:
+    5.     raise InvalidIssuanceRequestProof
+
+    // Create BBS signature on (c, k, r)
+    6. e <- Zq
+    7. X_A = G + H1 * c + K
+    8. A = X_A * (1/(e + sk))
+    9. X_G = G * (e + sk)
+
+    // Generate proof of correct computation
+   10. relation = build_relation_issue_response(A, X_A, G, X_G)
+   11. nizk = NISigmaProtocol("response", relation)
```

Comment:
This is probably neither here nor there, but the previous code went with "respond". Perhaps @SamuelSchlesinger has an opinion here

Reply:
you are right.